packages/@aws-cdk/aws-s3-assets/lib/asset.ts

```diff
@@ -191,6 +207,8 @@ export class Asset extends CoreConstruct implements cdk.IAsset {
     // points to a local path in order to enable local invocation of this function.
     resource.cfnOptions.metadata = resource.cfnOptions.metadata || { };
     resource.cfnOptions.metadata[cxapi.ASSET_RESOURCE_METADATA_PATH_KEY] = this.assetPath;
+    resource.cfnOptions.metadata[cxapi.ASSET_RESOURCE_METADATA_ORIGINAL_PATH_KEY] = this.originalAssetPath;
```

Comment:
Are we injecting absolute paths into templates? Doesn't that break build determinism across machines?

See https://github.com/cdklabs/construct-hub/pull/648/files#r764446322 (copy @Chriscbr).

Reply:
I think we can't afford for this to be an absolute path because it means that CDK output will differ between systems (e.g. where the project directory is, etc).

Reply:
Tracking the fix here: https://github.com/aws/aws-cdk/issues/17706